GOFevaluation/utils.py

```diff
@@ -162,42 +198,49 @@ def _get_equiprobable_binning(reference_sample, n_partitions, order=None):
         Reference: F. James, 2008: "Statistical Methods in Experimental
                     Physics", Ch. 11.2.3
     """
+    if reference_sample_weights is None:
+        weights_flag = 0
+    else:
+        mesg = 'data and their weights should be in the same length'
+        assert len(reference_sample) == len(reference_sample_weights), mesg
```

Comment:
very nice! I tried to test a few more things to throw the algorithm off 😈 and so far, negative weights are allowed. Maybe this should also be checked? I'm also fine with leaving this to the responsibility of the user but otherwise, I would suggest writing a function let's say `_check_weight_sanity`, which just makes this assertion of the correct len plus some more things like no negative values and that it's one-dimensional etc. 
Not a necessity I would say, just a suggestion for you to consider. I personally like code that complains/warns you when you want to use it in a non-intended way 😊

Reply:
Sure. Will add it.